boot.js

```diff
@@ -131,6 +131,7 @@ function Boot (server, opts, done) {
 
   Plugin.loadPlugin.call(this, main, (err) => {
     debug('root plugin ready')
+    this.emit('preReady')
```

Comment:
Should we run this if we have errors? Can you add some unit tests to specify that case as well?

Reply:
Since is *pre* I would say yes. I'll add some test.